src/Command/MailerCitizenProjectValidatedSummaryCommand.php

```diff
@@ -0,0 +1,155 @@
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $adherents = $this->adherentRepository->findAllEnabledAdherents();
+
+        $counter = 0;
+        foreach ($adherents as $chunk) {
+            /** @var Adherent $adherent */
+            $adherent = $chunk[0];
+
+            // Finds all citizen projects near the Adherent
+            $citizenProjects = $this->citizenProjectRepository->findNearByCitizenProjectSummariesForAdherent(
+                $adherent,
+                self::CITIZEN_PROJECTS_LIST_LIMIT,
+                $input->getOption('maxAge')
+            );
+
+            if (0 === count($citizenProjects)) {
+                continue;
+            }
+
+            // Broadcast the citizen projects and returns them formatted in array
+            $broadcastedCitizenProjects = $this->broadcast($citizenProjects);
+
+            // Sends the message
+            $message = $this->createMessage($broadcastedCitizenProjects, $adherent);
+            $this->mailer->sendMessage($message);
+
+            $this->logger->info(
+                sprintf(
+                    '[Broadcasting citizen projects] An email with %d CP has been queued for adherent #%d',
+                    count($citizenProjects),
+                    $adherent->getId()
+                )
+            );
+            ++$counter;
+        }
+        $this->logger->info(sprintf('[Broadcasting citizen projects] Total emails queued : %d', $counter));
+    }
+
+    /**
+     * Puts the citizen projects broadcast flag and returns the list.
+     *
+     * @param array $citizenProjects List of CitizenProject
+     *
+     * @return array List of citizen projects formatted for the template
+     */
+    private function broadcast(array $citizenProjects): array
+    {
+        $broadcastedCitizenProjects = [];
+
+        try {
+            $this->entityManager->beginTransaction();
+
+            /** @var CitizenProject $citizenProject */
+            foreach ($citizenProjects as $citizenProject) {
+                if ($citizenProject->isBroadcasted()) {
+                    continue;
+                }
+
+                $citizenProject->setBroadcasted(true);
+                $this->entityManager->persist($citizenProject);
+
+                array_push($broadcastedCitizenProjects, $citizenProject->getSummary());
+            }
+
+            $this->entityManager->flush();
+            $this->entityManager->commit();
+
+            return $broadcastedCitizenProjects;
+        } catch (Exception $e) {
+            $this->logger->error(
+                '[Broadcasting citizen projects] An error occurred while broadcasting',
+                ['exception' => $e]
+            );
+        }
+    }
+
+    private function createMessage(
+        array $broadcastedCitizenProjects, Adherent $adherent): CitizenProjectApprovedSummaryMessage
```

Comment:
CS

Reply:
This method should be moved in a message factory. So a broadcaster could also depend on it.